webpack/assets/javascripts/react_app/components/ColumnSelector/ColumnSelector.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import { PropTypes } from 'prop-types';
+import {
+  Button,
+  Select,
+  SelectOption,
+  SelectVariant,
+} from '@patternfly/react-core';
+
+const ColumnSelector = props => {
+  const {
+    data: { url, controller, selected, columns },
+  } = props;
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedColumns, setSelectedColumns] = useState(selected);
+
+  const onToggle = isExpanded => {
+    setIsOpen(isExpanded);
+  };
+
+  const onSelect = (event, selection) => {
+    if (selectedColumns.find(i => i === selection)) {
+      setSelectedColumns(selectedColumns.filter(e => e !== selection));
+    } else {
+      setSelectedColumns([...selectedColumns, selection]);
+    }
+  };
+
+  const onChange = event => {
+    const allColumns = [...columns, 'Select All'];
+    if (JSON.stringify(selectedColumns) === JSON.stringify(allColumns)) {
```

Comment:
Could we do set operations here? This is ordering-dependant and that could cause issues if someone created table preferences through hammer (like I did)